pallets/claims/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+use codec::Encode;
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	dispatch::DispatchResult,
+	ensure,
+	traits::Get,
+	weights::{DispatchClass, Pays},
+};
+use frame_system::ensure_signed;
+use hex::FromHex;
+use orml_traits::{MultiCurrency, MultiCurrencyExtended};
+use orml_utilities::with_transaction_result;
+use primitives::{Amount, AssetId, Balance, CORE_ASSET_ID};
+use sp_io::{crypto::secp256k1_ecdsa_recover, hashing::keccak_256};
+use sp_runtime::traits::Zero;
+use sp_std::prelude::*;
+use sp_std::vec::Vec;
+
+pub use traits::*;
+
+mod claims_data;
+mod traits;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+mod benchmarking;
+
+use weights::WeightInfo;
+pub mod weights;
+
+pub trait Config: frame_system::Config {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+	type Currency: MultiCurrencyExtended<Self::AccountId, CurrencyId = AssetId, Balance = Balance, Amount = Amount>;
```

Comment:
I was just thinking that there is no need to use multi currency in this pallet. Claims only deal with native/core asset. So we could just use the Currency from the frame support. And then use balances pallet to handle native currency. 

just for info: multi currency depends on the native currency "balances" anyway. And if you deposit core asset like you do here in this pallet - the multi currency just calls the balances deposit. So we can get rid of the multi currency layer and call balances directly.

Reply:
I was thinking similarly and originally used the currency type. But when talking about DEX, it just seemed more natural and correct to use the multicurrency type and explicitly define that it is the core asset we want to deposit even though it is not required here. It is also aligned with other pallets. Shall we modify it regardless?

Reply:
Yes, this makes sense too. But i wonder if using multi currency support here would require also adding additioan tests to make sure that we do not deposit to any other currency ( might sound farfetched?!, not sure). I will discuss this with @jak-pan and we'll see.

Reply:
Your call. 

We can simplify to be quick and safe. 
If we actually do this in a generic way. We can probably pr this to orml? It would need more testing tho.

First approach is more general. Second approach is more advanced and we're limiting users so that they need to use orml. 

Q to think about: Can this be a configurable trait? Can we do without id if no asset_id is present and have it if it is present?

Reply:
As a fan of the Occam's razor, I would choose quick and safe. I should be ready with a commit tomorrow.